algorithms task (2)/student/include/impl.h

```diff
@@ -9,4 +9,7 @@
  *
  * @return true if source starts from a prefix
  */
-bool startsWith(const std::string& source, const std::string& prefix);
+bool startsWith(const std::string& source, const std::string& prefix)
+{
+    return std::equal(prefix.begin(), prefix.end(), source.begin());
```

Comment:
what if source less than prefix?

Reply:
According to cpp reference it will just return false:
1,3) Returns true if the range [first1, last1) is equal to the range [first2, first2 + (last1 - first1)), and false otherwise

Reply:
right, thanks 👍

Reply:
range1: 12345
range2: 12
first1: 1
last1: 6
first2: 1
last2: 6